integrations/gitlab_v2/main.py

```diff
@@ -0,0 +1,157 @@
+import typing
+from enum import StrEnum
+from typing import Any
+
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+from client import CREATE_UPDATE_WEBHOOK_EVENTS, DELETE_WEBHOOK_EVENTS, GitlabClient
+from integration import GitlabProjectResourceConfig
+from utils import extract_issue_payload, extract_merge_request_payload
+
+
+class InvalidTokenException(Exception):
+    ...
+
+
+class ResourceKind(StrEnum):
+    GROUP = "group"
+    PROJECT = "project"
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+class WebHookEventType(StrEnum):
+    MERGE_REQUEST = "merge_request"
+    ISSUE = "issue"
+
+
+# Listen to the start event of the integration. Called once when the integration starts.
+@ocean.on_start()
+async def on_start() -> None:
+    logger.info("Starting musah_gitlab integration")
+    tokens = ocean.integration_config["gitlab_access_tokens"]["tokens"]
+
+    # Check if tokens is a list and filter valid tokens (strings only)
+    if isinstance(tokens, list):
+        tokens_are_valid = filter(lambda token: isinstance(token, str), tokens)
+
+        # Ensure all tokens are valid strings
+        if not all(tokens_are_valid):
+            raise InvalidTokenException("Invalid access tokens, ensure all tokens are valid strings")
+    else:
+        raise InvalidTokenException("Invalid access tokens, confirm you passed in a list of tokens")
+
+    return await bootstrap_client()
+
+
+def initialize_client(gitlab_access_token: str) -> GitlabClient:
+    return GitlabClient(
+        ocean.integration_config["gitlab_host"],
+        gitlab_access_token,
+    )
```

Comment:
why do you pass the access token to this function? why can't you do the same as the gitlab_host?

Reply:
The access token is passed to the function because it can vary dynamically depending on the request, as my implementation supports multiple access tokens.  This allows the flexibility to switch between tokens as needed. In contrast, the GitLab host is a constant value that doesn’t change, so it's directly accessed from the configuration rather than passed as a parameter, making the token the only part that requires flexibility during initialization.

Reply:
Please let me know if there is no need to support multiple access tokens

Reply:
supporting multiple tokens is one of the main focus of this assignment so yeah, good job on that. I was only concerned about the way you've structured it